src/clients/QBittorrent.ts

```diff
@@ -312,61 +380,55 @@ export default class QBittorrent implements TorrentClient {
 						save_path: path,
 						isComplete: true,
 						autoTMM: false,
-						category: linkingCategory,
+						category: linkCategory,
 					}
-				: await this.getTorrentConfiguration(searchee);
-
-			const newCategoryName =
-				duplicateCategories && searchee.infoHash
-					? await this.setUpCrossSeedCategory(category)
-					: category;
+				: await this.getTorrentConfiguration(
+						searchee as SearcheeWithInfoHash,
+					);
+
+			const tags = await this.getTagsForNewTorrent(searchee, {
+				save_path,
+				isComplete,
+				autoTMM,
+				category,
+			});
 
 			if (!isComplete) return InjectionResult.TORRENT_NOT_COMPLETE;
-
-			const contentLayout =
-				!path &&
-				newTorrent.isSingleFileTorrent &&
-				(await this.isSubfolderContentLayout(searchee))
+			const contentLayout = path
+				? "Original"
+				: (await this.isSubfolderContentLayout(
+							searchee,
+							(await this.getTorrentInfo(searchee.infoHash!))[0],
+					  ))
 					? "Subfolder"
 					: "Original";
-
-			const skipRecheckTorrent =
-				decision === Decision.MATCH_PARTIAL ? skipRecheck : true;
-
 			const formData = new FormData();
 			formData.append("torrents", buffer, filename);
-			formData.append("tags", TORRENT_TAG);
-			formData.append("category", newCategoryName);
 
-			if (autoTMM) {
-				formData.append("autoTMM", "true");
-			} else {
-				formData.append("autoTMM", "false");
-				formData.append("savepath", save_path);
-			}
 			if (path) {
-				formData.append("contentLayout", "Original");
-				formData.append("skip_checking", skipRecheck.toString());
-				formData.append("paused", (!skipRecheck).toString());
-			} else {
-				formData.append("contentLayout", contentLayout);
-				formData.append("skip_checking", skipRecheckTorrent.toString());
-				formData.append("paused", (!skipRecheckTorrent).toString());
+				// we were provided a path, so set it
+				formData.append("savepath", save_path);
 			}
 
+			formData.append(
+				"autoTMM",
+				flatLinking && searchee.infoHash ? autoTMM.toString() : "false",
+			);
+			formData.append("contentLayout", path ? "Original" : contentLayout);
```

Comment:
looks like this could be simplified as we're already checking for path higher up
```suggestion
			formData.append(contentLayout);
```